nbs/02_match_class.ipynb

```diff
@@ -224,7 +224,9 @@
     "        self.graph: DiGraph = graph\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #10.            # self._collections: List[CollectionName] = List() # Collections Feature
</pre><p>i dont see why you need both the list of names and the dict. The dict's keys act as the list of names, no?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
<p>We initially thought that it should be implemented the same way as nodes and edges - their names are saved in a list, and their values (mapping to the actual nodes and edges in the input graph) are saved in a separate member.</p><p>This can be removed as the dictionary saves their names as keys.</p>

Reply:
<pre class="ql-syntax" spellcheck="false"><code></code>` class CollectionView(set):
      def __init__(self,set)
        self.set=set
      def __getitem__(self,attr):
        if attr.in_index_like():
          return self.set[attr]
        return {i.get(attr) for i in self.set if attr in i }
</pre><p> </p><p> </p><p> </p>